### DISCUSSION TITLE
Enable format for yamlls language server

### DISCUSSION THREAD

**Discussion by robsonpeixoto** [LazyVim/LazyVim#730] [Q&A]
Even configuring `format = { enable = true }`, the `Format Document` does not appear in Code actions list:

![image](https://user-images.githubusercontent.com/124390/236877357-fc58f353-cbd7-4f7c-8daf-78ed7bead633.png)

Any idea of how to add it?

```lua
return {
  -- add json to treesitter
  {
    "nvim-treesitter/nvim-treesitter",
    opts = function(_, opts)
      if type(opts.ensure_installed) == "table" then
        vim.list_extend(opts.ensure_installed, { "yaml" })
      end
    end,
  },

  {
    "neovim/nvim-lspconfig",
    dependencies = {
      "b0o/SchemaStore.nvim",
      version = false, -- last release is way too old
    },
    opts = {
      -- make sure mason installs the server
      servers = {
        yamlls = {
          -- lazy-load schemastore when needed
          on_new_config = function(new_config)
            new_config.settings.yaml.schemas = new_config.settings.yaml.schemas or {}
            vim.list_extend(new_config.settings.yaml.schemas, require("schemastore").yaml.schemas())
          end,
          settings = {
            yaml = {
              format = { enable = true },
              validate = { enable = true },
              schemas = require("schemastore").yaml.schemas(),
            },
          },
        },
      },

    },
  },
}

```

**Comment by Jomik** [comment]
Here is my config
https://github.com/Jomik/dotfiles/blob/718c8f82963dfc0f0a1c4c25725b75223024a68c/.config/nvim/lua/plugins/lang/yaml.lua
I think your `settings` for `yamlls` is invalid, and thus it doesn't apply any of them.
For `yamlls` it is `validate = true`, not with the object. 
See this https://github.com/redhat-developer/yaml-language-server#language-server-settings
You also don't need to send in the schemas as that is what your `on_new_config` is doing.

It should be 
```lua
          settings = {
            yaml = {
              format = { enable = true },
              validate = true,
            },
          },
```

**Reply by robsonpeixoto** [reply]
Tanks @Jomik . 
More context about this problem: https://github.com/redhat-developer/yaml-language-server/issues/486#issuecomment-1046792026
